index.js

```diff
@@ -25,7 +25,12 @@ function compress(input, params, cb) {
     process.nextTick(cb, new Error('Second argument is not a function.'));
     return;
   }
-  encode.compressAsync(input, params, cb);
+  var stream = new TransformStreamEncode(params);
+  var chunks = [];
+  stream.on('error', cb);
+  stream.on('data', function(c) { chunks.push(c); });
+  stream.on('end', function() { cb(null, Buffer.concat(chunks)); });
```

Comment:
According to the [docs](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_concat_list_totallength):

>If totalLength is not provided, it is calculated from the Buffers in the list. This, however, adds an additional loop to the function, so it is faster to provide the length explicitly.

I haven't benchmarked myself but I assume it'd be wise to count the length every time we push a new chunk.
This also applies to the following streams obviously.